RELEASE-NOTES.md

```diff
@@ -7,6 +7,130 @@ The format of this file is based on [Keep a Changelog](http://keepachangelog.com
 
 The Axom project release numbers follow [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [Version 0.4.0] - Release date 2020-09-22
+
+### Added
+- Exposed the tolerance parameter `EPS` that is used to determine intersections between
+  triangles in `primal:intersect()` as an optional final parameter.
+- Added BVH spatial index option to the `mesh_tester` utility for calculating
+  triangle-triangle intersection.
+- Added `axom::execution_space< ExecSpace >::onDevice()` to check if execution
+  space is on device.
+- Added Axom macro `AXOM_SUPPRESS_HD_WARN` to silence host device compiler
+  warnings.
+- Added option to quest's `SignedDistance` class and C API to toggle whether
+  the distance query computes the sign.
+- Added a `batched` option to quest's signed distance query example application.
+  This computes all distance queries on an array of points using a single call to `computeDistance`.
+  The query uses OpenMP threading, when available.
+- Added new component, Inlet, to assist in retrieving and storing data from
+  an input deck.
+- Added the ability to specify an [Umpire] allocator ID to use with the
+  BVH. This allows the application to use a device allocator for the BVH and 
+  avoid use of UM on the GPU, which can hinder perfomrmance, or use a pool
```

Comment:
Suggestion: UM -> "Unified Memory (UM)" ?  (in case a reader is not familiar with that acronym)